File: app/src/main/java/ru/samtakoy/core/model/utils/MyStringUtils.java
package ru.samtakoy.core.model.utils;

import java.util.Iterator;

public class MyStringUtils {


    private static final String EMPTY = "";

    public static String join(final Iterable<?> iterable, final String separator) {
        return join(iterable.iterator(), separator);
    }

    public static String join(final Iterator<?> iterator, final String separator) {

        // handle null, zero and one elements before building a buffer
        if (iterator == null) {
            return null;
        }
        if (!iterator.hasNext()) {
            return EMPTY;
        }
        final Object first = iterator.next();
        if (!iterator.hasNext()) {
            return String.valueOf(first);
        }

        // two or more elements
        final StringBuilder buf = new StringBuilder(); // Java default is 16, probably too small
        if (first != null) {
            buf.append(first);
        }

        while (iterator.hasNext()) {
            if (separator != null) {
                buf.append(separator);
            }
            final Object obj = iterator.next();
            if (obj != null) {
                buf.append(obj);
            }
        }
        return buf.toString();
    }

}

File: app/src/main/kotlin/ru/samtakoy/core/app/utils/MyStringUtils.kt
package ru.samtakoy.core.app.utils

object MyStringUtils {
    private const val EMPTY = ""

    fun join(iterable: Iterable<*>, separator: String): String {
        return MyStringUtils.join(iterable.iterator(), separator)
    }

    fun join(iterator: Iterator<*>, separator: String): String {
        // handle null, zero and one elements before building a buffer
        if (!iterator.hasNext()) {
            return EMPTY
        }
        val first = iterator.next()
        if (!iterator.hasNext()) {
            return first.toString()
        }

        // two or more elements
        val buf = StringBuilder() // Java default is 16, probably too small
        if (first != null) {
            buf.append(first)
        }

        while (iterator.hasNext()) {
            if (separator != null) {
                buf.append(separator)
            }
            val obj = iterator.next()
            if (obj != null) {
                buf.append(obj)
            }
        }
        return buf.toString()
    }
}
